packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -66,7 +67,16 @@ module.exports = {
   // You can exclude the *.map files from the build during deployment.
   devtool: shouldUseSourceMap ? 'source-map' : false,
   // In production, we only want to load the polyfills and the app code.
-  entry: [require.resolve('./polyfills'), paths.appIndexJs],
+  entry: {
+    // Load the app and all its dependencies
+    main: paths.appIndexJs,
+    // Add the polyfills
+    polyfills: require.resolve('./polyfills'),
+    // List of all the node modules that should be excluded from the app
+    vendors: fs.existsSync(paths.appVendors)
+      ? require(paths.appVendors)
+      : ['react', 'react-dom'],
```

Comment:
@gaearon would you agree these are decent defaults, or would you rather disable the vendor bundle all together?

I'm not sure how many people use `react-scripts` without `react`. 😄

Reply:
can you use react-scripts without react ?
If you want to use it with preact for instance, you have to eject so...

Reply:
You wouldn't have to eject if you don't need to alias (external modules), right?

Reply:
I've never used external modules so I have no ideas

Reply:
I meant a `node_module` by external modules.

Reply:
But the external modules have to be configure in Webpack I think (in the external field)

Reply:
oh, yes it's true. I haven't thought of this, you can also do this with babel

But if we want to disable the possibility to have no vendors, I cannot set the array to an empty one, an entry has to be unempty, so I'll have to add or not the `vendors` entry

Reply:
We can wait for @gaearon's input, but I think we'd just change:
```js
  entry: {
    // Load the app and all its dependencies
    main: paths.appIndexJs,
    // Add the polyfills
    polyfills: require.resolve('./polyfills'),
    // List of all the node modules that should be excluded from the app
    vendors: fs.existsSync(paths.appVendors)
      ? require(paths.appVendors)
      : ['react', 'react-dom'],
  },
```

to something like this:

```js
       entry: Object.assign(
        {
          // Load the app and all its dependencies
          main: paths.appIndexJs,
          // Add the polyfills
          polyfills: require.resolve('./polyfills'),
        },
        fs.existsSync(paths.appVendors)
          ? { vendors: require(paths.appVendors) }
          : {}
      ),
```

Reply:
You're right, it makes more sense to only load the vendors if the file exists, it's more understandable this way

Reply:
@Timer, I managed to import the entries in the right order with `HtmlWebpackPlugin` and the option `chunksSortMode`
`
What wasn't working before is that I've imported an entry before the entry called runtime.
And if you place the entries here in a wrong order, it won't work. The right order is `['runtime', 'vendors', 'polyfills', 'main']`
But as webpack loads the entries without any bugs, I think it auto detects which entries must be imported before the others so I don't know if it's really usefull to hardcode the order of the entries

Reply:
Well if it's not needed we can remove it, but only if we're sure!

Reply:
I'm not sure at all, it's just that if the order is wrong, it won't compile and this is the only arrangement that works, so it can't be by chance.

I've added the commit anyways, thus if we want to remove it, it's pretty easy and the work isn't lost